packages/drivers/odsp-driver/src/odspDriverUrlResolver2.ts

```diff
@@ -104,13 +106,14 @@ export class OdspDriverUrlResolver2 implements IUrlResolver {
 
         const odspResolvedUrl = await new OdspDriverUrlResolver().resolve(requestToBeResolved);
 
-        // Kick start the sharing link request if we don't already have it already as a performance optimization.
-        // For detached create new, we don't have an item id yet and therefore cannot generate a share link
-        if (!isSharingLink && odspResolvedUrl.itemId) {
-            sharingLinkP = this.getShareLinkPromise(odspResolvedUrl);
+        if (isRedeemSharingLink) {
+            odspResolvedUrl.redeemSharingLink = request.url.split("?")[0];
         }
-        if (sharingLinkP) {
-            odspResolvedUrl.sharingLinkP = sharingLinkP;
+        if (odspResolvedUrl.itemId) {
+            // Kick start the sharing link request if we don't already have it already as a performance optimization.
+            // For detached create new, we don't have an item id yet and therefore cannot generate a share link
+            const key = this.getKey(odspResolvedUrl);
+            this.sharingLinkCache.add(key, async () => this.getShareLinkPromise(odspResolvedUrl));
```

Comment:
Doesn't this.getShareLinkPromise already handle putting this in the cache?

Reply:
yeah. Removed here.